controller/src/main/java/io/pravega/controller/server/bucket/ZooKeeperBucketManager.java

```diff
@@ -122,9 +148,109 @@ public CompletableFuture<Void> initializeBucket(int bucket) {
         return bucketStore.createBucket(getServiceType(), bucket);
     }
 
+    @SneakyThrows(Exception.class)
```

Comment:
Per our code guidelines https://github.com/pravega/pravega/wiki/Contributing
this is not an acceptable use of SneakyThrows. If you are doing this for the interface, please convert them to a subclass of RuntimeException for the generic errors or for cases where the error is not real narrow the sneakyTHrows declaration to a specific type.

Reply:
Basically here `cache.start(true);` (Which is method of NodeCache class of curator framework) is throwing checked exception.
```
/**
     * Same as {@link #start()} but gives the option of doing an initial build
     *
     * @param buildInitial if true, {@link #rebuild()} will be called before this method
     *                     returns in order to get an initial view of the node
     * @throws Exception errors
     */
    public void     start(boolean buildInitial) throws Exception 
```
I just sneakyThrow the same.

Reply:
Yes. The curator framework's code quality is low. But we shouldn't lower our standards just because we are interacting with it. Currently `bucketStore` already has this issue. It internally uses `ZKStoreHelper` which handles this problem. If you look at it's code you can see in each case it is catching the exception and doing something like: `StoreException.create(StoreException.Type.UNKNOWN, e, path)` to wrap it. The same should be done here.

Reply:
Done as suggested.